src/rust/engine/options/src/lib.rs

```diff
@@ -158,27 +159,62 @@ pub struct OptionParser {
 
 impl OptionParser {
     pub fn new(env: Env, args: Args) -> Result<OptionParser, String> {
+        let buildroot = BuildRoot::find()?;
+        let buildroot_string = String::from_utf8(buildroot.as_os_str().as_bytes().to_vec())
+            .map_err(|e| {
+                format!(
+                    "Failed to decode build root path {}: {}",
+                    buildroot.display(),
+                    e
+                )
+            })?;
+
+        let mut seed_values = HashMap::from([
+            ("buildroot".to_string(), buildroot_string.clone()),
+            ("homedir".to_string(), shellexpand::tilde("~").into_owned()),
+            ("user".to_string(), whoami::username()),
```

Comment:
There is a slight difference here between the Python and Rust implementations:

Python uses `getpass.getuser()`, which looks at $LOGNAME,$USER,$LNAME,$USERNAME and then the password file, and then we fall back to the stringified uid if no username is found (e.g., when running in a docker container with a host uid).

The whoami crate looks just at the password file, and falls back to the string "unknown".

I think that in both fallback cases it doesn't matter what we pick, it's not going to correspond to a real username, and we pick something just so the code can proceed. In fact "unknown" is probably a better choice.

It's possible we should consult those env vars here as well, but in practice I suspect it doesn't matter at all on a modern system with a password file, and there is no need to drag that legacy behavior on.